projects/plugins/jetpack/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -927,7 +977,7 @@ public function update_data( $request ) {
 
 				default:
 					// If option value was the same, consider it done.
-					$updated = get_option( $option ) != $value ? update_option( $option, $value ) : true;
+					$updated = get_option( $option ) !== $value ? update_option( $option, $value ) : true;
```

Comment:
Same here, if `$value` is a non-string scalar it'll wind up being stringified by `update_option()`. Should we keep the loose comparison, or do `$value = is_scalar( $value ) ? (string) $value : $value;` first, or something like that?

Reply:
👍 74547b3574d2da24d72b5cfb40b1068d3f945b3c